lib/ProductOpener/Tags.pm

```diff
@@ -318,6 +319,22 @@ sub has_tag($$$) {
 	return $return;
 }
 
+# Helper function to tell if a product has a certain tag from the passed list
+sub has_one_of_the_tags_from_the_list {
+	
+	my($product_ref, $tagtype, $tag_list_ref) = @_;
+	
+	my $flag = 0;
+
+	foreach my $tag_name (@$tag_list_ref) {
+			if ( has_tag($product_ref, $tagtype, $tag_name) ) {
+				$flag = 1;
+				return $flag;
```

Comment:
It looks good, but if you have a return statement here, we don't need a $flag variable, you can just put "return 1;" here, and "return 0;" at the end of the function.

Or if we want to have only 1 return statement at the end, you could replace "return $flag;" by "last;" to exit from the loop.